Update docs with new Healthcheck information (#296)
hazelcast/management-center-docs: docs/modules/clusters/pages/healthcheck.adoc

= Healthcheck
:description: You can run Healthcheck to analyze member configs and metrics in a cluster.
:page-aliases: ROOT:healthcheck.adoc
:page-enterprise: true

{description}

image:ROOT:Healthcheck.png[Healthcheck]

Healthcheck provides various checks related to configuration semantics, environment, member config equality, and usage.

A new Healthcheck is performed upon navigation to the Healthcheck page. You can run all checks again using the `RUN HEALTHCHECK` button in the header.

== Healthcheck Status

> Healthcheck categorizes the checks and problems with five different severities:

Reviewer: I think severities are only FAIL/WARNING/INFO which indicate `non pass` status.
PASS and DISABLED are two other types of status

Author: I changed the sentence. Can you please take a look at it again?

> - *FAIL:* A check has the status `FAIL`, indicated by a red color, when the cluster exhibits at least one critical problem, signaling the need for urgent resolution to prevent system failure or compromise.

Reviewer: nit: I think `system failure` is a too strong wording here. For instance mismatching map config doesn't lead to cluster shutdown, but to an unpredicted behaviour and possibly functional problems. 
@jgardiner68 do you have an idea how to phrase it correctly?

Author: I changed it with "when the cluster exhibits at least one critical problem that may result in an unexpected behavior or an issue with the designed functionality." WDYT?

== Actions
image:ROOT:HealthcheckActions.png[Healthcheck Actions]

Healthcheck provides several actions to be taken using the Actions dropdown in the header. These actions include the following:

- *Run Full Healthcheck:* When a regular Healthcheck is performed, disabled checks are typically ignored. However, this action specifically allows for the execution of a Healthcheck that includes the results of disabled checks.

NOTE: After running a Full Healthcheck, preview mode is enabled. When Healthcheck goes into preview mode, the user is not able to disable any check. Running regular Healthcheck disables the preview mode.

- *Download Report:* This action can be used to download the Healthcheck report in JSON format, for further analysis.
- *Reset Healthcheck Settings:* This action removes all information about disabled checks and data structures. After resetting the settings, Healthcheck will include all checks and data structures in the analysis.

== Filters
image:ROOT:HealthcheckFilter.png[Healthcheck Filters]
Healthcheck provides a mechanism to filter the results based on several attributes. These filters include the following:

- *Search:*

> Healthcheck results can be filtered by searching for check ID, description, problematic member, and data structure.

Author: I made the change, but used "or" instead of "and/or". We choose only one of these options.

- *Status Filter:*
You can keep track of the number of checks based on their statuses. Additionally, you can toggle the statuses on/off to filter the results in the report by clicking on the status buttons.
- *Category Filter:*
Healthcheck categorizes the checks by several attributes. The results can be filtered by enabling/disabling the checks in different categories, by using choose category dropdown in the header.

NOTE: ID of a check has a letter prefix about the related category.

If at least one filter is in use, all the filters can be cleared using the `Reset Filters` button.

== Healthcheck Problems
image:ROOT:HealthcheckProblems.png[Healthcheck Problems]

Individual check results can be inspected by clicking on the report item associated with a specific check. A report item may contain several problems of different statuses. A problem summary consists of a problem description

> and advice to fix the issue.

Author: "advice" is a special term in Healthcheck. I used "and an advice to fix the issue"

== Disabling Checks

> A check can be disabled when it is not critical, and omitting the check is feasible for a cluster. When a check is disabled, Healthcheck does not perform an operation about this particular check.

Reviewer 2: ```suggestion
Only disable a check when it is not critical and when omitting that check is viable for a cluster. When a check is disabled, Healthcheck does not perform that check.
```

> Healthcheck provides two disabling mechanisms:

Reviewer 2: ```suggestion
You can disable Healthcheck in either of the following ways:
```

Author: I added this one, as "You can disable a check in either of the following ways:".
Healthcheck is a general term.

> *Disabling for the whole cluster:* A check can be disabled for the whole cluster by clicking on the checkbox in the details of a report item. Cluster-wide disabled check will not be performed on any member or data structure.

Reviewer 2: ```suggestion
* *Disable for the whole cluster:* Select the checkbox in the details of a report item to disable the check for the whole cluster . If selected, the check is not performed on any member or data structure in the cluster.
```

> *Disabling for a specific data structure:* A check can be disabled for a specific data structure by clicking on the `Disable For` button inside a problem detail. Note that the check will be performed on the other data structures,

Reviewer 2: ```suggestion
* *Disable for a specific data structure*: Select the *Disable For* button in the problem details to disable the check for a specific data structure. 
+
NOTE: The other data structures are checked,
```

> and it may still contain problems about other data structures.

Reviewer 2: ```suggestion
and can identify problems with other data structures including the one for which the check has been disabled.
```

Author: I guess this "Note that" part is not needed completely :/ I removed it.

> 

Reviewer 2: ```suggestion
+
```

Author: Is this a leftover?

Reviewer 2: I had suggested making the points a list - to keep the note in-line with the list item you need to connect the lines with a plus. It is just formatting for the compiler.

If not making them a list, you don't need the plus.

> NOTE: Disabling for a specific data structure works only for specific checks. This mechanism does not exist for every check.

Reviewer: This mechanism does not exist for every check. => There is no mechanism to disable all checks for a particular data structure.

Author: Changed it with the suggestion.

> NOTE: Disabling for a specific data structure works only for specific checks. This mechanism does not exist for every check.

Reviewer 2: ```suggestion
You can only disable a specific check for a data structure; you cannot disable all checks for that data structure.
```

Author: What I meant here is closer to the Pavlo's suggestion here: https://github.com/hazelcast/management-center-docs/pull/296#discussion_r1486248363
I changed it.

> Disabling a check causes the corresponding report item to move down the list. The details of a disabled check looks like the following:

Reviewer 2: ```suggestion
When a check is disabled, the corresponding report item moves down the list. The details of a disabled check are similar to the following:
```

image:ROOT:HealthcheckEnable.png[Healthcheck Enable]

> A check may be further enabled for the whole cluster, or for a specific data structure from the report details of the disabled check.

Reviewer 2: ```suggestion
You can enable a check for the whole cluster, or for a specific data structure, from the report details of the disabled check.
```

Reviewer 2: Just a question: 

I presume that historic check reports are retained, and this is how you can select a report for a disabled check. If the check is disabled before Healthcheck is ever run (if this is possible), there would be no report.

Is there any situation in which a disabled check would not have an existing report through which it can be re-enabled?

Author: We don't retain historical check reports. We maintain a fixed check list, and keep record of the disabled check list of the user. So, whenever the user gets a Healthcheck report, they get all of the disabled checks and report items, through which they can re-enable the check.

Reviewer 2: Fantastic, and thank you for explaining.

> - If a check is disabled cluster-wide, the checkbox inside the report details can be used to enable the check for the whole cluster.

Reviewer 2: ```suggestion
- If a check is disabled cluster-wide, deselect the checkbox in the report details to enable the check for the whole cluster.
```

> - If a check is disabled for a specific data structure, details about the disabled data structures is shown in the report item. You can further enable the check for a specific data structure by clicking on the `Enable For` button inside

Reviewer 2: ```suggestion
- If a check is disabled for a specific data structure, select the *Enable For* button in the report details to enable the check for that data structure.
```

> the problem details.

Reviewer 2: ```suggestion

```
